daq/stream_monitor.py

```diff
@@ -32,7 +35,9 @@ def monitor(self, name, desc, callback=None, hangup=None, copy_to=None, error=No
             self.make_nonblock(desc)
             callback = lambda: self.copy_data(name, desc, copy_to)
         LOGGER.debug('Monitoring start %s fd %d', name, fd)
-        self.callbacks[fd] = (name, callback, hangup, error)
+        timeout = datetime.fromtimestamp(time.time()) + \
```

Comment:
extract second clause to variable -- same number of lines and avoids line continuation

delta = timedelta(seconds=timeout_sec) if timeout_sec else None
timeout = datetime.fromtimestamp(time.time()) + delta

If that's the right logic -- that's another reason -- the precedence between "+" and "if" is not obvious.

Reply:
I think I'll use traditional if statement to make it clearer.